### PR TITLE
fix hpu destructors flow and remove finish_measurements

### PR DESCRIPTION
**Pull request by nirda7**
Fix hpu related destructors flow, avoiding multiple calls to the same shutdown function.
Remove the use of the finish_measurements function.

### CODE REVIEW

**Review by Tiefen-boop — DISMISSED** [review]

File: vllm/executor/ray_hpu_executor.py [hunk 1]
```diff
@@ -70,7 +70,12 @@ def _init_executor(self) -> None:
         self.output_decoder = msgspec.msgpack.Decoder(
             Optional[List[SamplerOutput]])
 
+        self.shutdown_inc = True
+
     def shutdown(self) -> None:
+        if self.shutdown_inc:
```
Comment (kzawora-intel) [inline vllm/executor/ray_hpu_executor.py:76]:
```suggestion
        if getattr(self, 'shutdown_inc', False):
```

File: vllm/executor/hpu_executor.py [hunk 2]
```diff
@@ -197,7 +195,9 @@ def stop_profile(self) -> None:
         self.driver_worker.stop_profile()
 
     def shutdown(self) -> None:
-        self.driver_worker.shutdown_inc()
+        if self.shutdown_inc:
```
Comment (kzawora-intel) [inline vllm/executor/hpu_executor.py:198]:
```suggestion
        if getattr(self, 'shutdown_inc', False):
```

**Review by kzawora-intel — COMMENTED** [review]
LGTM overall, but let's add a guard against `AttributeError` in the destructor by using getattr (in case executor's destructor gets called before `_init_executor()`)

File: vllm/executor/hpu_executor.py (hunk 2 above)
Comment (nirda7) [reply vllm/executor/hpu_executor.py]: Done

File: vllm/executor/ray_hpu_executor.py (hunk 1 above)
Comment (nirda7) [reply vllm/executor/ray_hpu_executor.py]: Done.

**Review by afierka-intel — APPROVED** [review]
LGTM

**Review by michalkuligowski — APPROVED** [review]

**Pull request merged**